packages/api/cms-api/src/mailer/mailer.service.ts

```diff
@@ -0,0 +1,55 @@
+import { Inject, Injectable, Logger } from "@nestjs/common";
+import { Transporter } from "nodemailer";
+import { Address, Options as MailOptions } from "nodemailer/lib/mailer";
+
+import { MAILER_MODULE_OPTIONS, MAILER_MODULE_TRANSPORT } from "./mailer.constants";
+
+type MailerServiceConfig = {
+    defaultFrom: string;
+    sendAllMailsTo?: string[];
+    sendAllMailsBcc?: string[];
+};
+
+@Injectable()
+export class MailerService {
+    private readonly logger = new Logger(MailerService.name);
+
+    constructor(
+        @Inject(MAILER_MODULE_OPTIONS) private readonly mailerConfig: MailerServiceConfig,
+        @Inject(MAILER_MODULE_TRANSPORT) private readonly mailerTransport: Transporter,
+    ) {}
+
+    private fillMailOptionsDefaults(originMailOptions: MailOptions): MailOptions {
+        return {
+            ...originMailOptions,
+            from: originMailOptions.from || this.mailerConfig.defaultFrom,
+            bcc: this.mailerConfig.sendAllMailsBcc
+                ? [...this.normalizeToArray(originMailOptions.bcc), ...this.mailerConfig.sendAllMailsBcc]
+                : originMailOptions.bcc,
+        };
+    }
+
+    /**
+     * Sends a mail and logs it in the database.
+     * @param type Mail type, e.g. order confirmation, order cancellation, etc. to filter in the mailer log
+     * @param additionalData Put your additional data here, e.g. orderId, resourcePoolId, etc.
+     * @param originMailOptions `from` defaults to this.config.mailer.defaultFrom, sendAllMailsBcc is always added to `bcc`
+     */
+    async sendMail({ type, additionalData, ...originMailOptions }: MailOptions & { type?: string; additionalData?: unknown }) {
+        const mailOptionsWithDefaults = this.fillMailOptionsDefaults(originMailOptions);
+
+        // this is needed because only on production stage we are allowed to send mails to customers
+        const mailOptions: MailOptions = this.mailerConfig.sendAllMailsTo
+            ? { ...mailOptionsWithDefaults, to: this.mailerConfig.sendAllMailsTo, cc: undefined, bcc: undefined }
+            : mailOptionsWithDefaults;
+
+        const result = await this.mailerTransport.sendMail(mailOptions);
+        if (!result.messageId) throw new Error(`Sending mail failed, no messageId returned. MailOptions: ${JSON.stringify(mailOptions)}`);
```

Comment:
Does the result contain an error? If yes, the error is currently silenced.

Which errors are caught here? If suppose the library throws if e.g. the SMTP server is not available

Reply:
To be honest I don't know which error nodemailer handles and how. But I started some research on this...

maybe I should also stringify the result into the error message? but this will also be logged (follow up PR: https://github.com/vivid-planet/comet/pull/3967/files#diff-60c828cb3070c8e48683c0d10e66cabdb5f18aa73b67e36c1da6a1f6b4d610b9R71)